gtsfm/loader/mobilebrick_loader.py

```diff
@@ -0,0 +1,152 @@
+        self._num_images = len(os.listdir(self._image_dir))
+
+        # Cache image paths
+        self._image_paths = []
+        for i in range(self._num_images):
+            image_path = os.path.join(self._image_dir, f"{i:06d}.jpg")
+            self._image_paths.append(image_path)
+
+        # Load GT intrinsics
+        if self._use_gt_intrinsics:
+            intrinsics_dir = os.path.join(data_dir, "intrinsic")
+            self._intrinsics = []
+            for i in range(self._num_images):
+                intrinsics_file = os.path.join(intrinsics_dir, f"{i:06d}.txt")
+                K = np.loadtxt(intrinsics_file)
+                self._intrinsics.append(Cal3Bundler((K[0, 0] + K[1, 1]) / 2, 0, 0, K[0, 2], K[1, 2]))
+        else:
+            # TODO(akshay): It should be possible to cache approx intrinsics here.
+            self._intrinsics = None
+
+        # Load GT poses
+        self._poses_dir = os.path.join(data_dir, "pose")
+        self._wTi = []
+        for i in range(self._num_images):
+            pose_file = os.path.join(self._poses_dir, f"{i:06d}.txt")
+            wTi_mat = np.loadtxt(pose_file)
+            wTi = Pose3(Rot3(wTi_mat[:3, :3]), wTi_mat[:3, 3])
+            self._wTi.append(wTi)
+
+    def image_filenames(self) -> List[Path]:
+        """Return the file names corresponding to each image index."""
+        return [Path(fpath) for fpath in sorted(os.listdir(self._image_dir))]
+
+    def __len__(self) -> int:
+        """The number of images in the dataset.
+
+        Returns:
+            The number of images.
+        """
+        return self._num_images
+
+    def get_image_full_res(self, index: int) -> Image:
+        """Get the image at the given index, at full resolution.
+
+        Args:
+            index: The index to fetch.
+
+        Returns:
+            The image at the query index.
+
+        Raises:
+            IndexError: If an out-of-bounds image index is requested.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+
+        # Read in image.
+        img = io_utils.load_image(self._image_paths[index])
+        return Image(value_array=img.value_array, exif_data=img.exif_data, file_name=img.file_name)
+
+    def get_camera_intrinsics_full_res(self, index: int) -> Cal3Bundler:
+        """Get the camera intrinsics at the given index, valid for a full-resolution image.
+
+        Args:
+            index: The index to fetch.
+
+        Returns:
+            Ground truth intrinsics for the given camera.
+
+        Raises:
+            IndexError: If an out-of-bounds image index is requested.
+        """
+        if index < 0 or index >= len(self):
+            raise IndexError(f"Image index {index} is invalid")
+
+        if self._intrinsics:
+            return self._intrinsics[index]
+        else:
+            # 0.8 is better than the default factor of 1.2 for this dataset, but it has not been fully tuned.
```

Comment:
nit: could be helpful to add a TODO to use a system-wide default, instead of per-dataset options.

Reply:
The system-wide default is 1.2, but it would not work for all datasets. unless we change the verifier or refine intrinsics in BA. 
Are you suggesting adding a TODO to update this when we find a way to use a system-wide default?

Reply:
Yup, exactly